<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>565</height>
   </rect>
  </property>
  <property name="windowIcon">
   <iconset resource="../../presentation/grypto_ui.qrc">
    <normaloff>:/grypto/icons/main.png</normaloff>:/grypto/icons/main.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="page_locked">
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="btn_unlock">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>1</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
          <property name="whatsThis">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;The window is locked. You must enter the correct credentials to continue using this file.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Click to Unlock</string>
          </property>
          <property name="icon">
           <iconset resource="../../presentation/grypto_ui.qrc">
            <normaloff>:/grypto/icons/main.png</normaloff>:/grypto/icons/main.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>200</width>
            <height>200</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_unlocked">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="Grypt::EntryView" name="view_entry" native="true">
          <property name="whatsThis">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="menu_Recent_Files">
     <property name="title">
      <string>&amp;Recent Files</string>
     </property>
    </widget>
    <widget class="QMenu" name="menu_Export">
     <property name="title">
      <string>&amp;Export</string>
     </property>
     <addaction name="action_export_ps"/>
    </widget>
    <widget class="QMenu" name="menu_Import">
     <property name="title">
      <string>&amp;Import</string>
     </property>
     <addaction name="action_import_ps"/>
    </widget>
    <addaction name="actionNewOpenDB"/>
    <addaction name="action_Save_As"/>
    <addaction name="action_Close"/>
    <addaction name="separator"/>
    <addaction name="menu_Export"/>
    <addaction name="menu_Import"/>
    <addaction name="separator"/>
    <addaction name="actionLockUnlock"/>
    <addaction name="menu_Recent_Files"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEntry">
    <property name="title">
     <string>&amp;Entry</string>
    </property>
    <addaction name="actionNew_Entry"/>
    <addaction name="action_EditEntry"/>
    <addaction name="action_DeleteEntry"/>
    <addaction name="separator"/>
    <addaction name="action_Search"/>
   </widget>
   <widget class="QMenu" name="menu_Tools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="action_cryptoTransform"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="action_Undo"/>
    <addaction name="action_Redo"/>
    <addaction name="separator"/>
    <addaction name="action_Favorites"/>
    <addaction name="action_Preferences"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menuEdit"/>
   <addaction name="menuEntry"/>
   <addaction name="menu_Tools"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dw_treeView">
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;View the entries you have stored in the encrypted database. You can organize them into a hierarchy by dragging and dropping them. If you add an entry to favorites, it will appear in the tray icon context menu when you right-click. Use the search window to filter and search the entries.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Entries are stored securely in the encrypted database, and only decrypted when you open it with this application. Entries may contain a file, which is also encrypted securely in the database.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Password Database</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="GUtil::Qt::TreeView" name="treeView">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="dragEnabled">
        <bool>true</bool>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::InternalMove</enum>
       </property>
       <property name="defaultDropAction">
        <enum>Qt::MoveAction</enum>
       </property>
       <property name="uniformRowHeights">
        <bool>true</bool>
       </property>
       <property name="animated">
        <bool>true</bool>
       </property>
       <property name="expandsOnDoubleClick">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QDockWidget" name="dw_search">
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Use the search window to filter and highlight entries that match the search parameters in the tree view.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;You can use wildcards or regular expressions, filter by date/time, etc... This should make finding your secrets as easy as possible.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="windowTitle">
    <string>Search</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="Grypt::SearchWidget" name="searchWidget" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionNew_Entry">
   <property name="text">
    <string>&amp;New</string>
   </property>
  </action>
  <action name="action_Save_As">
   <property name="text">
    <string>&amp;Save As...</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Save the current file as a new one.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="action_Open">
   <property name="text">
    <string>&amp;Open Database</string>
   </property>
  </action>
  <action name="actionNewOpenDB">
   <property name="text">
    <string>New/&amp;Open Database</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Open a database, creating a new one if it doesn't exist.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="action_Close">
   <property name="text">
    <string>&amp;Close Database</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Close the file&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="action_EditEntry">
   <property name="text">
    <string>&amp;Edit</string>
   </property>
  </action>
  <action name="action_DeleteEntry">
   <property name="text">
    <string>&amp;Delete</string>
   </property>
  </action>
  <action name="action_Undo">
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="action_Redo">
   <property name="text">
    <string>&amp;Redo</string>
   </property>
  </action>
  <action name="action_Search">
   <property name="text">
    <string>&amp;Search</string>
   </property>
  </action>
  <action name="actionBla">
   <property name="text">
    <string>bla</string>
   </property>
  </action>
  <action name="actionLockUnlock">
   <property name="text">
    <string>&amp;Lock Application</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Locks the interface so nobody can use the file until they enter the right credentials.&lt;/p&gt;&lt;p&gt;You can adjust the application preferences to have it automatically lock after a certain time of inactivity.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="action_cryptoTransform">
   <property name="text">
    <string>Cryptographic &amp;Transformations</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;A utility that helps you apply cryptographic transformations to data, such as:&lt;/p&gt;&lt;p&gt;    * AES encryption with CCM, one of the most secure encryption algorithms&lt;/p&gt;&lt;p&gt;    * Hash calculator with all major hashes&lt;/p&gt;&lt;p&gt;    * Conversion to base-64 and base-16 strings&lt;/p&gt;&lt;p&gt;You can transform files directly or operate on strings. This utility is included as a supplement to the main application. It is not directly useful to the main function, but it can come in handy when you need to check a hash.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="action_export_ps">
   <property name="text">
    <string>&amp;Portable Safe</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Export all entries and files to a secure Portable Safe file (.gps).&lt;/p&gt;&lt;p&gt;The Portable Safe file, invented by George Karagoulis, securely stores and obscures everything about the data it contains. It is impossible to determine the file's structure or glean any information about the contents without possessing the user's password and/or keyfile. It is the most secure way to transport your data if you have to expose it to the outside world, such as during a file transfer.&lt;/p&gt;&lt;p&gt;After you export, you can import it by creating a new empty file and saying Import-&amp;gt;Portable Safe.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="action_import_ps">
   <property name="text">
    <string>&amp;Portable Safe</string>
   </property>
   <property name="whatsThis">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Import entries and files from a Portable Safe file (.gps)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="action_Preferences">
   <property name="text">
    <string>&amp;Preferences</string>
   </property>
  </action>
  <action name="action_Favorites">
   <property name="text">
    <string>Organize &amp;Favorites</string>
   </property>
  </action>
  <action name="action_add_remove_favorites">
   <property name="text">
    <string>&amp;Add to Favorites</string>
   </property>
  </action>
  <action name="action_What_s_This">
   <property name="text">
    <string>&amp;What's This?</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Grypt::SearchWidget</class>
   <extends>QWidget</extends>
   <header>grypto_searchwidget.h</header>
   <container>1</container>
   <slots>
    <signal>FilterChanged(Grypt::FilterInfo_t)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>GUtil::Qt::TreeView</class>
   <extends>QTreeView</extends>
   <header>gutil/treeview.h</header>
  </customwidget>
  <customwidget>
   <class>Grypt::EntryView</class>
   <extends>QWidget</extends>
   <header>grypto_entryview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../presentation/grypto_ui.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>treeView</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>MainWindow</receiver>
   <slot>_treeview_doubleclicked(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>186</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>370</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchWidget</sender>
   <signal>FilterChanged(Grypt::FilterInfo_t)</signal>
   <receiver>MainWindow</receiver>
   <slot>_filter_updated(Grypt::FilterInfo_t)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>542</y>
    </hint>
    <hint type="destinationlabel">
     <x>322</x>
     <y>472</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_unlock</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>RequestUnlock()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>433</x>
     <y>213</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>_treeview_clicked(QModelIndex)</slot>
  <slot>_treeview_doubleclicked(QModelIndex)</slot>
  <slot>_tableview_doubleclicked(QModelIndex)</slot>
  <slot>_filter_updated(Grypt::FilterInfo_t)</slot>
  <slot>RequestUnlock()</slot>
  <slot>_export_file()</slot>
 </slots>
</ui>
